draft-ietf-oauth-device-flow.xml

```diff
@@ -385,42 +392,51 @@
+      the verification URI (<spanx style="verb">verification_uri</spanx>), 
+      as this increases the length and complexity of the URI that the user must type.
+      The next section documents user interaction with <spanx style="verb">verification_uri_complete</spanx>
+      which is designed to carry this information.
   </t>
+
+  <section title="Non-textual Verification URI Optimization" anchor="verification-optimizations">
   <t>
-  <figure><artwork><![CDATA[
-  https://example.com/device?user_code=WDJB-MJHT
-]]></artwork></figure>
+    When <spanx style="verb">verification_uri_complete</spanx> is included
+    in the Authorization Response (<xref target="AuthorizationResponse"/>),
+    clients MAY present this URI in a non-textual manner using
+    any method that results in the browser being opened with the URI, such as with
+    QR codes or NFC, to save the user typing the URI.
   </t>
   <t>
-    When the user code is included in the verification URI in this way, it is
-    considered as a hint to the authorization server to enable potential
-    optimizations. The authorization server MAY
-    use this hint to optimize the user interaction (such as by removing the need
-    for the user to type the code), it MAY also ignore it completely. The client
-    MUST still display the user code textually, for authorization servers that
-    require users to input the user code manually, or otherwise use the user
-    code as part of a visual confirmation step.
+    For usability reasons,
+    it is RECOMMENDED for clients to still display the
+    textual verification URI (<spanx style="verb">verification_uri</spanx>) for
+    users not able to use such a shortcut. Clients MUST
+    still display the <spanx style="verb">user_code</spanx>, as the authorization
+    server may still require the user to confirm it to disambiguate devices,
+    or as a remote phishing mitigation (See <xref target="RemotePhishing"/>).
   </t>
+        <t>
+              <figure anchor="user_instruction_with_optimzation"
+         title="Example User Instruction With QR Code Representation of the Complete Verification URI">
+               <artwork><![CDATA[
+               +-------------------------------------------------+
+               |                                                 |
+               |  Using a browser on another     +------------+  |
+               |  device, visit:                 |[_]..  . [_]|  |
+               |  https://example.com/device     | .  ..   . .|  |
+               |                                 | . .  . ....|  |
+               |                                 |.   . . .   |  |
+               |  And enter the code:            |[_]. ... .  |  |
+               |  WDJB-MJHT                      +------------+  |
+               |                                                 |
+               +-------------------------------------------------+
```

Comment:
Did you really intend to have that many spaces at the start of the lines in the diagram.  Typically these will have no leading spaces.  Please delete the leading spaces within the <artwork> block.